### PR TITLE
Updated main changelog

### PR DESCRIPTION
**Pull request by b-quachtran**
**Proposed changes**:
- Updated `main` branch changelog

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: CHANGELOG.mdx [hunk 1]
```diff
@@ -1729,6 +1748,27 @@ https://github.com/RasaHQ/rasa/tree/main/changelog/ . -->
...
+- [#8429](https://github.com/rasahq/rasa/issues/8429): Added `partition_by_sender` flag to Kafka Producer to optionally associate events with Kafka partition based on sender_id.
+
+### Improvements
+- [#8345](https://github.com/rasahq/rasa/issues/8345): Improved the [lock store](lock-stores.mdx) debug log message when the process has to queue because other messages have to be processed before this item.
+
+
+## [1.10.24] - 2021-03-29
+
+### Bugfixes
+- [#8019](https://github.com/rasahq/rasa/issues/8019): Added `group_id` parameter back to `KafkaEventBroker` to fix error when instantiating event broker with a config containing the `group_id` parameter which is only relevant to the event consumer
```
Comment (wochinge) [inline CHANGELOG.mdx:1769]: thanks for doing these old ones as well 💯

**Review by wochinge — COMMENTED** [review]
Thanks for taking care of this! 💯 

I think [the changelog for 2.5](https://github.com/RasaHQ/rasa/pull/8901) is missing, isn't it?

**Comment by b-quachtran** [comment]
@mvielkind FYI, added the 2.5.X changelog here as well

**Review by wochinge — APPROVED** [review]
🙌🏻

**Pull request merged**